### ISSUE TITLE
Remove ambiguity about harmony/kind == "none" vs. harmony/root-step

### ISSUE THREAD

**Issue by infojunkie** [w3c/musicxml#349]
As per [this forum discussion](https://forums.makemusic.com/viewtopic.php?f=12&t=2476#p9099), specifying `harmony/kind = "none"` is a signal to stop the harmony at the current event, but the current MusicXML spec still requires a `harmony/root-step`, which confuses music software systems that import or export MusicXML. It would be great to tie the `harmony/root-step` requirement to a value of `harmony/kind` different from `"none"`.

**Comment by MarcSabatella** [comment]
The context here, I think, is looking for a good way to represent "N.C." as a chord symbol.  I guess we could special-case text="" on the root-step as a command to disable all chord symbol recognition, but this really doesn't seem right to me.  Do we know for sure this is how real-world applications export "N.C."?

**Comment by infojunkie** [comment]
> Do we know for sure this is how real-world applications export "N.C."?

From my limited research based on immediate needs, MuseScore does the following:
```
      <harmony>
        <root>
          <root-step text="">C</root-step>
          </root>
        <kind text="N.C.">none</kind>
        </harmony>
```

and iReal Pro does the following:
```
      <harmony>
        <root>
          <root-step text="">C</root-step>
        </root>
        <kind text="N.C.">none</kind>
      </harmony>
```

which seems pretty consistent to me.

**Comment by mdgood** [comment]
Thanks for raising this @infojunkie. Yes, the way that MuseScore and iReal Pro are working is what is intended, and also matches how Finale works. I thought we had documented this already, but I see that I was mistaken. I think it would indeed be a good idea to clarify the documentation for this.

We can't really change the root-step requirement given our use of XSD and DTD technology. We would need something like Schematron for that, which as far as I know is still not that widely implemented. However we can clarify that for a chord with a kind of "none", the root-step and root-alter elements are ignored semantically, and the text attribute of the root-step element should be set to the empty string.

**Comment by MarcSabatella** [comment]
So does that mean you feel a kind of none with a given text should display only that text, regardless of the root-step value or its text?  What about a root-step text of "" but with an otherwise valid kind?

**Comment by mdgood** [comment]
@MarcSabatella This wouldn't change any current behavior. You still need to supply the text attributes for the different elements to control precise formatting. When a text attribute is specified for either the `<root-step>` or the `<kind>` element, that specifies what is displayed in that part of the chord symbol. 

What I think would be good to clarify is that *semantically* the root or function element is ignored when the `<kind>` element has a value of none, and that in most such cases you want to specify `<root-step text="">`. This would not take away the ability to specify the root-step text for other `<kind>` values.

I'm not sure if I am answering your question though - please let me know if I am missing something.

**Comment by MarcSabatella** [comment]
I guess I wasn't clear on the distinction between ignoring something *semantically* versus ignoring it *literally*.  So to ask a very specific question, if in the original example shown above from MuseScore and iReal Pro, the text="" was omitted from the root-step tag, then the proper display would be "CN.C.", but it should be treated as not a valid harmony element, and thus should not be treated as a "C" chord of any kind of with regard to transposition, playback, etc?  It also seems to follow from what you say that the text="N.C." could reasonably be added to *either* the root-step *or* the kind tags, as long as the value of kind was none, and the result would be semantically the same and should also display the same - as "N.C.".

**Comment by mdgood** [comment]
@MarcSabatella, the N.C. text could indeed go in either place and product a technically correct result. I think it best for it to go with the `<kind>` element since it is a specific visual representation of the `none` value. It also matches application common practice. I think it would be good to more precisely document this recommended practice.

**Comment by mdgood** [comment]
@infojunkie, could you please review #367 to see if that resolves both this issue and #352? Thanks!

**Comment by infojunkie** [comment]
Thanks, I left a comment on the PR.
